tensorflow/core/util/gpu_kernel_helper.h

```diff
@@ -314,6 +314,29 @@ int64_t MinAlignmentOf(Args... args) {
   return std::min({alignment_of(args)...});
 }
 
+namespace detail {
+
+template <int64_t VecSize, template <int vec_size> class Functor>
+struct DispatchToVectorizedHelper {
+  template <typename... Args>
+  Status operator()(int64_t max_vec_size, Args&&... args) const {
+    if (max_vec_size >= VecSize) {
+      return Functor<VecSize>()(std::forward<Args>(args)...);
+    }
+    return DispatchToVectorizedHelper<VecSize / 2, Functor>()(
+        max_vec_size, std::forward<Args>(args)...);
+  }
+};
+template <template <int vec_size> class Functor>
+struct DispatchToVectorizedHelper<0, Functor> {
```

Comment:
I think it's clearer for the base case to be VecSize=1 instead of VecSize=0, since I think the VecSize=0 case will never actually be executed.

Reply:
I've made this change. I got confused for a bit because it turns out there is a tested path that calls this with `max_vec_size=0`. That was happening because it was calling `MinAlignmentOf(nullptr, ...)` due to its input pointer being unused. I've fixed that too by treating zero/nullptr as infinitely aligned. I also added some error checks.